File: iteration2/src/CommandLineInterface/CLILogin.java
package iteration2.src.CommandLineInterface;
import java.util.Scanner;
import iteration2.src.controllers.UserController;
import iteration2.src.models.User;
//import controllers.UserController;
import iteration2.src.utils.Util;

public class CLILogin {

    private User user;

    

    public User loginPage(){
        Util.clearScreen();
        Scanner input = new Scanner(System.in);
        do {
            System.out.println(" Login ");
            System.out.println(" ******** ");

            System.out.print("Username:");
            String username = input.nextLine();

            System.out.print("Password:");
            String password = input.nextLine();

            //UserController userController = new UserController();
            UserController userController = new UserController();
            user = userController.loginUser(username,password);


            if(user == null){
                System.out.println("Invalid username or password.");
            }
        }
      //recursive yapma, do while a dönüştür
        while(user == null );

        //input.close();
        return user;
        }

    }

File: iteration3/src/CommandLineInterface/CLILogin.py
from iteration3.src.controllers.UserController import UserController
from iteration3.src.interfaces.Color import Color
from iteration3.src.models.User import User
from iteration3.src.utils.Util import Util

class CLILogin(object):
     
     def __init__(self):
        self._user = None


     def loginPage(self):
         while True:
            Util.clearScreen()
            print(" Login ")
            print(" ******** ")

            username = input("Username: ")
            password = input("Password: ")

            userController = UserController()
            self._user = userController.login(username, password)
            
            if self._user is None:
               Util.sendFeedback("Invalid username or password", Color.RED)

            else:
               break
            
         return self._user